db/db_impl/db_impl_open.cc

```diff
@@ -582,21 +582,41 @@ Status DBImpl::Recover(
               "existing log file: ",
               file);
         } else {
-          logs.push_back(number);
+          log_files[number] =
+              LogFileName(immutable_db_options_.wal_dir, number);
         }
       }
     }
 
-    if (logs.size() > 0) {
+    if (immutable_db_options_.track_and_verify_wals_in_manifest) {
+      // Verify WALs in MANIFEST.
+      s = versions_->GetWalSet().CheckWals(env_, log_files);
+      if (!s.ok()) {
+        return Status::Corruption("Failed to verify WALs tracked in MANIFEST",
```

Comment:
`CheckWals(...)` can return `IOError` if `GetFileSize` encounters an `IOError`. Maybe just propagate the IO error to application.